### PR TITLE
[fix bug 1130419] Update fetch_bugs to use Bugzilla REST API.

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

**Comment by akatsoulas** [comment]
@glogiotatidis or @johngian  can you please r this?

**Comment by johngian** [comment]
Sure @akatsoulas

File: remo/remozilla/tasks.py [hunk 1]
```diff
@@ -50,22 +51,37 @@ def fetch_bugs(components=COMPONENTS, days=None):
     Bugzilla users with users on this website, when possible.
 
     """
+    login_url = LOGIN_URL.format(username=settings.REMOZILLA_USERNAME,
+                                 password=settings.REMOZILLA_PASSWORD)
+    response = json.loads(requests.get(login_url).text)
```
Comment (glogiotatidis) [inline remo/remozilla/tasks.py]: if you're on a recent `requests` version, there is `requests.get(url).json`

File: remo/remozilla/tasks.py [hunk 2]
```diff
@@ -50,22 +51,37 @@ def fetch_bugs(components=COMPONENTS, days=None):
...
+    error = response.get('error', None)
```
Comment (glogiotatidis) [inline remo/remozilla/tasks.py]: `response.get` or `response.text.get`? https://github.com/mozilla/remo/pull/926/files#diff-3461c2b2e8b3521845166163c28bffe9R79

File: remo/remozilla/tasks.py [hunk 3]
```diff
@@ -24,12 +24,13 @@
...
-       'include_fields={fields}&changed_after={timedelta}d&'
-       'offset={offset}&limit={limit}')
-
+URL = 'https://bugzilla.mozilla.org/rest/bug'
+LOGIN_URL = ('https://bugzilla.mozilla.org/rest/login?login={username}'
+             '&password={password}')
+URL_PARAMS = ('&product=Mozilla%20Reps&component={component}&'
+              'include_fields={fields}&changed_after={timedelta}d&'
+              'offset={offset}&limit={limit}')
+TOKEN_PARAM = '?token={token}'
```
Comment (glogiotatidis) [inline remo/remozilla/tasks.py]: I'd merge this with URL_PARAMS because you're only using this together and you end up urlmerging this.

Comment (glogiotatidis) [inline remo/remozilla/tasks.py]: Even with URL as is was before.

**Comment by glogiotatidis** [comment]
I didn't actually run this but code looks ok.

**Comment by glogiotatidis** [comment]
BTW we have tests for this and they were not updated but they are not failing either. What's happening?

**Comment by akatsoulas** [comment]
@glogiotatidis pr updated

File: remo/remozilla/tests/test_tasks.py [hunk 4]
```diff
@@ -31,23 +32,20 @@ def test_connection_error(self, fake_get):
...
         """Test fetch_bugs invalid status code exception."""
         if ((not getattr(settings, 'REMOZILLA_USERNAME', None) or
              not getattr(settings, 'REMOZILLA_PASSWORD', None))):
             raise SkipTest('Skipping test due to unset REMOZILLA_USERNAME '
                            'or REMOZILLA_PASSWORD.')
-        request = requests.Request()
-        request.status_code = 500
-        request.text = 'Foobar'
-        fake_get.return_value = request
+        mocked_request.get.return_value = mocked_response = Mock()
```
Comment (glogiotatidis) [inline remo/remozilla/tests/test_tasks.py]: nah, no multiple assignments in one line please.

**Comment by glogiotatidis** [comment]
r+wc nice job!

**Pull request merged**